openstack/tools/charmed_openstack_functest_runner.sh

```diff
@@ -138,9 +138,19 @@ OAM_MAX=$(ipcalc $CIDR_OAM| awk '$1=="HostMax:" {print $2}')
 OAM_MIN=$(ipcalc $CIDR_OAM| awk '$1=="HostMin:" {print $2}')
 OAM_MIN_ABC=${OAM_MIN%.*}
 OAM_MAX_D=${OAM_MAX##*.}
-# Picking last two addresses and hoping they dont get used by Neutron.
-export {OS,TEST}_VIP00=${OAM_MIN_ABC}.$(($OAM_MAX_D - 1))
-export {OS,TEST}_VIP01=${OAM_MIN_ABC}.$(($OAM_MAX_D - 2))
+
+NUM_VIPS=2
+for ((i=$NUM_VIPS;i;i-=1)); do
+    vip_port_name=zaza-vip-port-$((i-1))
+    vip_addr=$(openstack port show -c fixed_ips $vip_port_name -f yaml| yq .fixed_ips[0].ip_address)
+    if [[ $vip_addr = null ]]; then
+        vip_addr=${OAM_MIN_ABC}.$(($OAM_MAX_D - $i))
```

Comment:
Does the IP address matter? It shouldn't, seems this could cause an issue if the IP is already in use.

Reply:
thats a good point, we could just use address now couldnt we since we are pre-allocating them. nice one brian.